### ISSUE TITLE
Update Customer Service Cards with Medallia Data

### ISSUE THREAD

**Issue by michelle-dooley** [department-of-veterans-affairs/va.gov-team#42080]
## Issue Description
The customer service cards on the standardized dashboards needs to be updated the reflect the most recent Medallia owned by contact center data similar to what we did for VAOS.

The contact center recently streamlined their process for updating the medallia data, we need to ensure we are pulling in based on any data structure changes.

---
## Tasks
- [x] Update customer service cards with new Medallia data
- [x] QA the new data source/cards
- [x] Publish

## Acceptance Criteria
- [x] When the above tasks are complete and the customer service cards on the standardized dashboards are reflective of the most recent Medallia data.
---
## How to configure this issue
- [x] **Attached to a Milestone** (when will this be completed?)
- [x] **Attached to an Epic** (what body of work is this a part of?)
- [x] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `Console-Services`, `tools-fe`)
- [x] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [x] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)

**Comment by chloedotbrown** [comment]
I've brought our product-to-URL tables for standardized dashboards into Domo to map onto the Medallia data already there:

1. [BQ - Standardized Dashboard Products - Forms](https://va-gov.domo.com/datasources/2da29c96-76e0-43f3-92ec-be1fa8c1f31f/details/overview) - connects to the `forms_steps` table in BQ, updates with a full replace weekly.
2. [BQ - Standardized Dashboard Products - Content](https://va-gov.domo.com/datasources/d1d35839-8a31-44d6-8410-c990cfb8b409/details/overview) - connects to the `content_urls` table in BQ, updates with a full replace weekly.
3. [Standardized Dashboard Products - Search](https://va-gov.domo.com/datasources/7b7d2370-dc6a-4587-939d-46b6cddba646/details/overview) - connects to a manually edited Google Sheet in our shared folder. If updated, must manually refresh dataset. Does not include 'VA Homepage' as search product, since we only have access to the URL where survey was submitted.

#### Next Steps

- [ ] Create new ETL that joins Medallia surveys to Forms, Content, and Search data tables
- [ ] Create 4 output datasets - one for each dashboard (all use different names for "product" field) and one that contains all Medallia data by known product URLs.
- [ ] QA output data: make sure total number of surveys per product, date range, and avg. ratings match base Medallia data already in Domo when filtered correctly. 

@jonathan-epstein13 , would you be down to do the QA step? I'll tag you in this ticket once data is ready.

**Comment by jonathan-epstein13** [comment]
@chloedotbrown sounds like a plan. Sign me up!

**Comment by chloedotbrown** [comment]
@jonathan-epstein13 - Medallia data for our 3 standardized dashboards is ready for your QA! Linking and details are below. 

- **[Forms Medallia Data](https://va-gov.domo.com/datasources/c5df2114-84ab-4375-81f1-f309aa513a65/details/overview)**: This dataset is created with the new `root_url` field I added to the forms definitions table in BigQuery. This is the base URL shared by all steps in the form, therefore catching customer surveys submitted on form steps we may not have on file. You can find the urls to filter by [here](https://va-gov.domo.com/datasources/2da29c96-76e0-43f3-92ec-be1fa8c1f31f/details/overview).

- **[Search Medallia Data](https://va-gov.domo.com/datasources/e799e52f-8502-4b5f-acb4-dbfbe65baf05/details/overview)**: This dataset is created using a similar url base pattern that shows up whenever someone submits a survey on a search results page. I manually added this to a Google Sheet, which is then used to maps search products from the Medallia data. You can find those url roots [here](https://va-gov.domo.com/datasources/7b7d2370-dc6a-4587-939d-46b6cddba646/details/overview). 

- **[Content Medallia Data](https://va-gov.domo.com/datasources/8c6474c6-c9e0-4b3e-babc-58480b33f6da/details/overview)**: This dataset is created using the specific urls included in the `content_urls` in BigQuery, which you can find in Domo [here](https://va-gov.domo.com/datasources/d1d35839-8a31-44d6-8410-c990cfb8b409/details/overview). These ought to be exact matches with those pages, with the exception of the VAMC and Vet Center pages, which have regex patterns in the urls. While Redshift does not allow a `regex_contains` match, I was able to recreate the join I needed with string operations, and surveys per product looked consistent when I did my own testing.

For all 3, the main thing that needs to be checked is whether all the surveys that should be included for each product have been captured in the new datasets. To QA, I'd recommend filtering by product in the new dataset and then using a `contains` filter on the underlying [Medallia data](https://va-gov.domo.com/datasources/3d83a71b-fba6-431d-8c80-5ce9d72312c2/details/data-lineage) to try to recreate the logic used to define the product (see each below). **If the row totals are the same, they should pass!** If they're off, let me know what patterns you're seeing and I'll investigate. Thanks for your help!

**Comment by pavanhothi** [comment]
@chloedotbrown @michelle-dooley I have created all the new Medallia cards for the standardized dashboards using the three new data sources Chloe created. @chloedotbrown not sure if you wanted to do a quick QA/scan of them before I went ahead and added them onto the dashboard! Thank you!!

**Comment by chloedotbrown** [comment]
@michelle-dooley - I've made edits and added cards to [Forms](https://va-gov.domo.com/page/447193050), [Search](https://va-gov.domo.com/page/1964748112), and [Content](https://va-gov.domo.com/page/426422632) dashboards. Do you want to give it a quick check on your end to make sure they look alright before we close ticket and announce to teams?

**Comment by michelle-dooley** [comment]
@chloedotbrown - everything looks great!! Couple things....

- The "Data Sources" listed at top of [search](https://va-gov.domo.com/page/1964748112) dashboard needs to be updated, still lists ForeSee and Medallia pending.  
- I noticed on the Content dashboard that for Campaign Landing Page, Translated Content and Virtual Agent Chatbot there was very little to no data, I assume that is by the nature of those products/filters and how we assign responses to each?

**Comment by chloedotbrown** [comment]
@michelle-dooley Good catch on the Search intro!! I've updated it and added a note that Medallia data isn't available for the homepage search.

As for the low Content totals, that's correct. I've added default filters to the bar charts to show the previous month's data only, but overall, the totals are just low here. They did pass row count comparisons with the Medallia base data in the data QA (Jonathan and I walked through together). Those URLs just seem to have comparatively low surveys on them.

Let me know if there's anything else you think should be changed?
